test-suite/buildtest.sh

```diff
@@ -22,19 +22,25 @@ if [ -n "$cache_file" ]; then
     configcache="--cache-file=$cache_file"
 fi
 
-#if we are on Linux, let's try parallelizing
+# let's try parallelizing
+if [ -z "$pjobs" ] && which nproc >/dev/null 2>&1 ; then
```

Comment:
The problem is not that running nproc cheap, but that it might not be installed.
nproc is part of GNU coreutils, and it's not a given it will be installed on a non-Linux system.
Your proposed approach would result in a shell-emitted error message about nproc not
being available - everything would work just fine, but the error message might be confusing

Reply:
The error message will not be visible to anybody AFAICT due to output redirection. Am I missing something?

Reply:
I had tested it. The error message was visible, I think because it is emitted by the shell itself, before output redirection was done

Reply:
This redirection is a standard shell trick we are using elsewhere (including PR code), but I tested it just to make sure. The error message is not visible in all cases I have tested, on two OSes (macOS and Linux):

```
% nproc
zsh: command not found: nproc

% if [ -z "$pjobs" ] && nproc > /dev/null 2>&1 ; then echo TRYING; else echo NOT TRYING; fi
NOT TRYING

% prjobs=12
% if [ -z "$pjobs" ] && nproc > /dev/null 2>&1 ; then echo TRYING; else echo NOT TRYING; fi
NOT TRYING
```

```
$ nproc
16

$ if [ -z "$pjobs" ] && nproc > /dev/null 2>&1 ; then echo TRYING; else echo NOT TRYING; fi
TRYING
$ if [ -z "$pjobs" ] && nprocXXX > /dev/null 2>&1 ; then echo TRYING; else echo NOT TRYING; fi
NOT TRYING

$ pjobs=12
$ if [ -z "$pjobs" ] && nprocXXX > /dev/null 2>&1 ; then echo TRYING; else echo NOT TRYING; fi
NOT TRYING
$ if [ -z "$pjobs" ] && nproc > /dev/null 2>&1 ; then echo TRYING; else echo NOT TRYING; fi
NOT TRYING
```

----

If you need more reasons to avoid `which` besides the original change request, some folks [assert](https://stackoverflow.com/a/677212) that "Many operating systems have a `which` that doesn't even set an exit status".